wit/metrics.wit

```diff
@@ -0,0 +1,258 @@
+interface metrics {
+    use wasi:clocks/wall-clock@0.2.0.{datetime};
+    use wasi:clocks/monotonic-clock@0.2.0.{duration};
+    use types.{key-value, instrumentation-scope};
+    use tracing.{span-id, trace-id};
+
+    /// `export` sends all metric data related to a Reader from the SDK
+    %export: func(metrics: resource-metrics) -> result<_, error>;
+
+    type error = string;
```

Comment:
nit: I question whether aliasing string to error is really any better 🤷? I don't have enough WIT experience to really know but I thought I'd raise the question and let you decide.